### PR TITLE
Use single quote

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | /
| Type?         | improvement
| Category?     | CO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | /
| How to test?  | /

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11740)

### CODE REVIEW

File: admin-dev/filemanager/include/utils.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ function foldersize($path)
     $cleanPath = rtrim($path, '/'). '/';
 
     foreach ($files as $t) {
-        if ($t<>"." && $t<>"..") {
+        if ($t<>'.' && $t<>'..') {
```
Comment (jolelievre) [inline admin-dev/filemanager/include/utils.php]: probably a conflict here

**Review by jolelievre — CHANGES_REQUESTED** [review]

File: admin-dev/filemanager/include/utils.php (hunk 1 above)
Comment (jolelievre) [reply admin-dev/filemanager/include/utils.php]: and other conflicts to fix from merged PRs
the rest is ok for me

File: var/SymfonyRequirements.php [hunk 2]
```diff
@@ -518,7 +518,7 @@ function_exists('simplexml_import_dom'),
         if (extension_loaded('suhosin')) {
             $this->addPhpIniRequirement(
                 'suhosin.executor.include.whitelist',
-                function ($cfgValue) { return false !== stripos($cfgValue, "phar"); },
+                function ($cfgValue) { return false !== stripos($cfgValue, 'phar'); },
```
Comment (Quetzacoalt91) [inline var/SymfonyRequirements.php]: Note this file may loose these changes at the first composer update.

**Review by Quetzacoalt91 — DISMISSED** [review]

**Comment by Quetzacoalt91** [comment]
Conflicts must be resolved before going further in the review process. :)

**Comment by MathiasReker** [comment]
@Quetzacoalt91 @jolelievre Conflicts solved.

File: var/SymfonyRequirements.php (hunk 2 above)
Comment (MathiasReker) [reply var/SymfonyRequirements.php]: Yes, I understand.

File: admin-dev/filemanager/execute.php [hunk 3]
```diff
@@ -34,12 +34,7 @@
 
 $base = $current_path;
 
-if (isset($_POST['path'])) {
-    $path = $current_path.str_replace("\0", '', $_POST['path']);
-} else {
-    $path = $current_path;
-}
-
+$path = isset($_POST['path']) ? ($current_path . $_POST['path']) : $current_path;
```
Comment (jolelievre) [inline admin-dev/filemanager/execute.php:37]: why change the original code? this str_replace is is used for legitimate safety reasons (null byte injection)
we recently fixed a few security breaches in this component, so I prefer not to modify its behavior if not necessary

File: admin-dev/filemanager/execute.php [hunk 4]
```diff
@@ -57,7 +52,7 @@
     $cycle = false;
 }
 
-$path = $current_path.str_replace("\0", '', $_POST['path']);
+$path = $current_path . $_POST['path'];
```
Comment (jolelievre) [inline admin-dev/filemanager/execute.php]: same thing here

**Review by jolelievre — CHANGES_REQUESTED** [review]
do not remove the null byte injection protection

**Comment by MathiasReker** [comment]
Sorry, I will be more careful next time I solve conflicts.

File: admin-dev/filemanager/execute.php [hunk 5]
```diff
@@ -33,7 +33,12 @@
 require_once $language_file;
 
 $base = $current_path;
-$path = isset($_POST['path']) ? ($current_path . $_POST['path']) : $current_path;
+
+if (isset($_POST['path'])) {
+    $path = $current_path.str_replace("\0", '', $_POST['path']);
```
Comment (PierreRambaud) [inline admin-dev/filemanager/execute.php]: You missed something here, this one should not be commit :) same below.

File: var/SymfonyRequirements.php (hunk 2 above)
Comment (PierreRambaud) [reply var/SymfonyRequirements.php]: You must revert changes here :/

Comment (MathiasReker) [reply var/SymfonyRequirements.php]: Okay, I will.

Comment (MathiasReker) [reply var/SymfonyRequirements.php]: Fixed

File: admin-dev/filemanager/execute.php (hunk 5 above)
Comment (MathiasReker) [reply admin-dev/filemanager/execute.php]: Thanks! Sorry, I messed a little up. Nice you catches my mistakes! :)

**Comment by MathiasReker** [comment]
All should be good now.

File: tools/parser_sql/utils/PHPSQLParserUtils.php [hunk 6]
```diff
@@ -92,33 +92,33 @@ protected function removeParenthesisFromStart($token) {
...
+            $trim[0] = ' ';
             $trim = trim($trim);
         }
 
         $parenthesis = $parenthesisRemoved;
         $i = 0;
         $string = 0;
         while ($i < strlen($trim)) {
+            if ($trim[$i] === '\\') {
+                $i += 2; # an escape character, the next character is irrelevant
```
Comment (jolelievre) [inline tools/parser_sql/utils/PHPSQLParserUtils.php]: `# -> //` fixed in https://github.com/PrestaShop/PrestaShop/pull/11699

File: admin-dev/filemanager/include/php_image_magician.php [hunk 7]
```diff
@@ -2497,7 +2497,7 @@ public function reset()
 
 ## --------------------------------------------------------
```
Comment (jolelievre) [inline admin-dev/filemanager/include/php_image_magician.php]: Some `#` left, I thought this was fixed in another PR that hase been merged https://github.com/PrestaShop/PrestaShop/pull/11699
problem during rebase maybe?

File: tools/parser_sql/processors/AbstractProcessor.php [hunk 8]
```diff
@@ -100,22 +100,23 @@ protected function removeParenthesisFromStart($token) {
         $string = 0;
         while ($i < strlen($trim)) {
 
-            if ($trim[$i] === "\\") {
-                $i += 2; // an escape character, the next character is irrelevant
+            if ($trim[$i] === '\\') {
+                $i += 2; # an escape character, the next character is irrelevant
```
Comment (jolelievre) [inline tools/parser_sql/processors/AbstractProcessor.php]: `# -> //`  fixed in https://github.com/PrestaShop/PrestaShop/pull/11699

File: tools/parser_sql/processors/CreateDefinitionProcessor.php [hunk 9]
```diff
@@ -119,8 +119,8 @@ public function process($tokens) {
                 break;
 
             case 'PRIMARY':
-                if ($prevCategory === "" || $prevCategory === "CONSTRAINT") {
-                    // next one is KEY
+                if ($prevCategory === '' || $prevCategory === 'CONSTRAINT') {
+                    # next one is KEY
```
Comment (jolelievre) [inline tools/parser_sql/processors/CreateDefinitionProcessor.php]: `# -> //` fixed in https://github.com/PrestaShop/PrestaShop/pull/11699

File: tools/parser_sql/processors/CreateDefinitionProcessor.php [hunk 10]
```diff
@@ -129,8 +129,8 @@ public function process($tokens) {
                 break;
 
             case 'UNIQUE':
-                if ($prevCategory === "" || $prevCategory === "CONSTRAINT") {
-                    // next one is KEY
+                if ($prevCategory === '' || $prevCategory === 'CONSTRAINT') {
+                    # next one is KEY
```
Comment (jolelievre) [inline tools/parser_sql/processors/CreateDefinitionProcessor.php]: `# -> //` fixed in https://github.com/PrestaShop/PrestaShop/pull/11699

File: tools/parser_sql/processors/SQLProcessor.php [hunk 11]
```diff
@@ -59,23 +59,23 @@ public function process($tokens) {
...
             }
 
             /*
              * If it isn't obvious, when $skip_next is set, then we ignore the next real token, that is we ignore whitespace.
              */
             if ($skip_next > 0) {
-                if ($trim === "") {
-                    if ($token_category !== "") { // is this correct??
+                if ($trim === '') {
+                    if ($token_category !== '') { # is this correct??
```
Comment (jolelievre) [inline tools/parser_sql/processors/SQLProcessor.php]: `# -> //` fixed in https://github.com/PrestaShop/PrestaShop/pull/11699

File: tools/parser_sql/processors/SQLProcessor.php [hunk 12]
```diff
@@ -59,23 +59,23 @@ public function process($tokens) {
...
                         $out[$token_category][] = $token;
                     }
                     continue;
                 }
-                //to skip the token we replace it with whitespace
-                $trim = "";
-                $token = "";
+                #to skip the token we replace it with whitespace
```
Comment (jolelievre) [inline tools/parser_sql/processors/SQLProcessor.php]: `# -> //` fixed in https://github.com/PrestaShop/PrestaShop/pull/11699

**Review by jolelievre — CHANGES_REQUESTED** [review]
Careful this PR introduces reverts for `# -> //` fixed in https://github.com/PrestaShop/PrestaShop/pull/11699
Maybe you didn't rebase develop? and no conflicts appeared

File: tools/parser_sql/processors/SQLProcessor.php (hunk 12 above)
Comment (MathiasReker) [reply tools/parser_sql/processors/SQLProcessor.php]: Fixed, but many of those comments should not be short comments. Thats why I would make an other PR for them :) But lets have them in this way for now. See latest commit.

**Comment by jolelievre** [comment]
Waow I didn't expect so many changes!! :P
You're some of these comments should use the block syntax, do you think it will be fixed in https://github.com/PrestaShop/PrestaShop/pull/11747 ? Or in another PR?

**Review by jolelievre — DISMISSED** [review]
Ok for me, waiting for a second approval from @PrestaShop/prestashop-core-developers

**Comment by MathiasReker** [comment]
> 
> 
> Waow I didn't expect so many changes!! :P
> You're some of these comments should use the block syntax, do you think it will be fixed in #11747 ? Or in another PR?

Exactly many of them need block syntax. I will fix it in an other PR. I would like to have  #11747 merged asap.

**Comment by MathiasReker** [comment]
Fixed conflicts

**Comment by MathiasReker** [comment]
Travis should pass now =)

**Comment by MathiasReker** [comment]
I don't understand why Travis CI does not pass?

**Comment by jolelievre** [comment]
@MathiasReker I reloaded it, seems like it's good!

File: tools/parser_sql/lexer/LexerSplitter.php [hunk 13]
```diff
@@ -53,7 +53,7 @@ class LexerSplitter {
 
     protected static $splitters = array('<=>', "\r\n", '!=', '>=', '<=', '<>', '<<', '>>', ':=', '\\', '&&', '||', ':=',
                                        '/*', '*/', '--', '>', '<', '|', '=', '^', '(', ')', "\t", "\n", "'", '"', '`',
-                                       ',', '@', ' ', '+', '-', '*', '/', ';');
+                                       ',', '@', ' ', '+', '-', '*', '/', ';');,
```
Comment (jolelievre) [inline tools/parser_sql/lexer/LexerSplitter.php]: not sure about this one..

**Review by jolelievre — CHANGES_REQUESTED** [review]

File: classes/helper/HelperForm.php [hunk 14]
```diff
@@ -26,8 +26,8 @@
  use PrestaShop\PrestaShop\Core\Addon\Module\ModuleManagerBuilder;
 
  /**
-  * @since 1.5.0
-  */
+ * @since 1.5.0
```
Comment (jolelievre) [inline classes/helper/HelperForm.php:29]: this alignment doesn't seem good.. did it come from php-cs-fixer?

**Review by jolelievre — CHANGES_REQUESTED** [review]

File: classes/helper/HelperForm.php (hunk 14 above)
Comment (MathiasReker) [reply classes/helper/HelperForm.php:29]: Yes, it did. I agree with you! But it solved the lint issue reported by travis.

File: tools/parser_sql/lexer/LexerSplitter.php (hunk 13 above)
Comment (MathiasReker) [reply tools/parser_sql/lexer/LexerSplitter.php]: yes, it looks weird. I will revert it.

Comment (MathiasReker) [reply tools/parser_sql/lexer/LexerSplitter.php]: Should be: `',', '@', ' ', '+', '-', '*', '/', ';',);`

Comment (MathiasReker) [reply tools/parser_sql/lexer/LexerSplitter.php]: and `);` on next line

Comment (MathiasReker) [reply tools/parser_sql/lexer/LexerSplitter.php]: Fixed

**Review by jolelievre — APPROVED** [review]
Requires another approval from @PrestaShop/prestashop-core-developers

File: classes/helper/HelperForm.php (hunk 14 above)
Comment (MathiasReker) [reply classes/helper/HelperForm.php:29]: I fixed it.

**Comment by matks** [comment]
Dear @MathiasReker, as discussed together and following our [Contributing rules](https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests-requirements) I must unfortunately close this PR as it is too large to be thoroughly reviewed. Thank you for the contribution.

**Pull request closed without merging**